### PR TITLE
Make logger configurable in the embedded `ferretdb` package

### LINKED ISSUE #4014: Make logger configurable in the embedded `ferretdb` package
### What should be done?

`ferretdb.Config` struct should include `Logger *zap.Logger`. If that field is set, `ferretdb.New` should use that logger. Otherwise, it should initialize logging as we do it now, but only once (using `sync.Once`). 

### Where?

https://github.com/FerretDB/FerretDB/blob/main/ferretdb/ferretdb.go

### Definition of Done

- `ferretdb` package updated;
- tests / examples updated;
- documentation comments updated.

### PR DESCRIPTION
**Pull request by fadyat**
# Description

Closes #4014.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [x] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [x] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: ferretdb/ferretdb.go [hunk 1]
```diff
@@ -214,21 +217,33 @@ func (f *FerretDB) MongoDBURI() string {
...
 	}
 
-	logging.Setup(l, "console", "")
-	logger = zap.L()
+	return getGlobalLogger()
+}
+
+// getGlobalLogger retrieves or creates a global logger using
+// a loggerOnce to ensure it is created only once.
+func getGlobalLogger() *zap.Logger {
```
Comment (fadyat) [inline ferretdb/ferretdb.go:237]: for tests need 'to create a separate file' or 'update the ferretdb_test.go file of the package' or 'export functions'

I don't like any of these approaches, they remain untested 😢

Comment (noisersup) [reply ferretdb/ferretdb.go:237]: It's okay, I don't think we need them here

**Review by noisersup — APPROVED** [review]
Good job! LGTM

File: internal/bson2/slog.go
Comment (AlekSi) [inline internal/bson2/slog.go:1]: Do we need changes in this file?

File: ferretdb/ferretdb.go [hunk 2]
```diff
@@ -214,21 +223,23 @@ func (f *FerretDB) MongoDBURI() string {
...
+	loggerOnce.Do(func() {
+		level := zap.ErrorLevel
+		if version.Get().DebugBuild {
+			level = zap.DebugLevel
+		}
+
+		logging.Setup(level, "console", "")
```
Comment (AlekSi) [inline ferretdb/ferretdb.go:240]: `logging.Setup` also adds a zap hook for `RecentEntries`. If we don't call this method, then `getLog` command handler will not work correctly

File: internal/bson2/slog.go
Comment (AlekSi) [reply internal/bson2/slog.go:1]: I can't confirm that locally. Are you on the same commit and liter versions? Please try `task init` to recompile them

Comment (fadyat) [reply internal/bson2/slog.go:1]: ok, I'll check this evening

**Review by chilagrow — APPROVED** [review]
Thanks for your contribution :hugs:

**Review by noisersup — APPROVED** [review]

**Pull request merged**